src/stories/Examples/LazyImageLoading/LazyImageLoading.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+  import Image from "$lib/components/Image";
+  import { RootIntersectionObserver } from "$lib/components/IntersectionObserver";
+
+  import sampleImage from "../../../sample.jpg";
+  import sampleImageBlurhash, {
+    width as sampleImageWidth,
+    height as sampleImageHeight,
+    mean as sampleImageMean,
+  } from "../../../sample.jpg?blurhash";
+
+  export let nativeLazyLoading = false;
+  export let intersectionObserverLazyLoading = true;
```

Comment:
I keep getting tripped up on the interaction between these two; only one of them gets applied in the end (either we let the browser handle it natively or we use the intersection observer), right? Would it be possible to add comments here that would show up in the story? Don't need a full logic table; just a "hey, the intersection observer doesn't do anything if we're using the native lazy loading" (assuming that's correct).

Alternatively, if the story isn't useful for anything other than showing the intersection observer, let's just disable the controls in the story file:
```js
parameters: {
  controls: {
    disable: true,
  }
}
```

Reply:
Good point on how the two booleans make it hard to see what's going on - I've refactored `Image` and `LazyImageLoading` to take an explicit lazy image loading type so we can debug it directly without remembering the right incantation of boolean settings for the thing we want to test. Can you take another look at it and let me know what you think of it now?